draft-hendrickson-privacypass-public-metadata-issuance.md

```diff
@@ -43,10 +46,220 @@ TODO Abstract
+
+# Issuance Protocol for Publicly Verifiable Tokens {#public-flow}
+
+This section describes a variant of the issuance protocol in {{Section 6 of PROTOCOL}}
+for producing publicly verifiable tokens including public metadata using {{PBLINDRSA}}.
+In particular, this variant of the issuance protocol works for the
+TODO_insert_variants of the blind RSA protocol variants described in {{Section 5 of BLINDRSA}}.
+
+The public metadata issuance protocol differs from the protocol in
+{{Section 6 of PROTOCOL}} in that the tokens (both before and after signing) contain metadata set by the client that is visible to the signer (issuer) and verifier. This means any Client can set metadata on a token. Any issuer or verifier may reject the token by aborting the protocol, but cannot set or mutate the metadata.
+
+Beyond this difference, the publicly metadata issuance protocol variant is
+nearly identical to the publicy verifiable issuance protocol variant. In
+particular, Issuers provide a Private and Public Key, denoted skI and pkI,
+respectively, used to produce tokens as input to the protocol. See
+{{Section 6.5 of PROTOCOL}} for how this key pair is generated.
+
+Clients provide the following as input to the issuance protocol:
+
+- Issuer Request URI: A URI to which token request messages are sent. This can
+  be a URL derived from the "issuer-request-uri" value in the Issuer's
+  directory resource, or it can be another Client-configured URL. The value
+  of this parameter depends on the Client configuration and deployment model.
+  For example, in the 'Split Origin, Attester, Issuer' deployment model, the
+  Issuer Request URI might be correspond to the Client's configured Attester,
+  and the Attester is configured to relay requests to the Issuer.
+- Issuer name: An identifier for the Issuer. This is typically a host name that
+  can be used to construct HTTP requests to the Issuer.
+- Issuer Public Key: `pkI`, with a key identifier `token_key_id` computed as
+  described in {{public-issuer-configuration}}.
+- Challenge value: `challenge`, an opaque byte string. For example, this might
+  be provided by the redemption protocol in [AUTHSCHEME].
+- Metadata value: `metadata`, an opaque byte string.
+
+Given this configuration and these inputs, the two messages exchanged in
+this protocol are described below. The constant `Nk` is defined as token type 0x1234 in
+{{PROTOCOL}}.
+
+## Client-to-Issuer Request {#public-request}
+
+The Client first creates an issuance request message for a random value
+`nonce` using the input challenge and Issuer key identifier as follows:
+
+~~~
+nonce = random(32)
+challenge_digest = SHA256(challenge)
+token_input = concat(0x1234, // Token type field is 2 bytes long
```

Comment:
This is so that the signature covers everything preceding the authenticator in Token. That said, it's redundant... since `Blind` will concatenate the input to the metadata, and could therefore be more expensive than desired.